week1/414_thrid_maximum.py

```diff
@@ -0,0 +1,16 @@
+# https://leetcode.com/problems/third-maximum-number/
+def third_maximum(nums: list[int]) -> int:
+    max_num = 3
+    num_list = sorted(set(nums), reverse=True)
```

Comment:
I also took this approach of using a sort, but actually is not optimal. sorting involves O(nlogn) time complexity, but this can actually be O(n) because you only need to find the maximum value three times, which for finding a maximum value is O(n) and you need to do that three times. the leetcode system and test case still accepts this solution, but for practicing purpose it's best that you try to pursue the optimal solution.